UiPath.FreeRdpClient/UiPath.SessionTools.Tests/ProcessRunnerTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Diagnostics;
+using System.Reflection;
+
+namespace UiPath.SessionTools.Tests;
+
+[Trait("Subject", nameof(ProcessRunner))]
+public class ProcessRunnerTests
+{
+    public static readonly EventWaitHandle Event = new EventWaitHandle(initialState: false, EventResetMode.ManualReset, name: "UiPathSessionTools_2a81d9cd60504d39b79347668186c615");
+
+    [Fact]
+    public async Task Run_ShouldProduceAConsistentReport_WhenCancellationOccurs()
+    {
+        const int Offset = 12345;
+        var timeout = TimeSpan.FromMilliseconds(500);
+
+        string fileName = Path.ChangeExtension(Assembly.GetExecutingAssembly().Location, ".exe");
+
+        ProcessRunner runner = new();
+        using CancellationTokenSource cts = new();
+
+        int currentPID;
+        using (var self = Process.GetCurrentProcess())
+        {
+            currentPID = self.Id;
+        }
+
+        var act = () => runner.Run(fileName, $"{Offset} {currentPID}", cts.Token);
+
+        _ = Task.Run(() =>
+        {
+            Event.WaitOne();
+            cts.CancelAfter(timeout);
+        });
```

Comment:
I would still go for something simpler like : 
`Task.Run("cmd", "/c \"echo line1&echo line2&timeout 1&echo line3\""`
and
```
stdout.shouldContain("line2")
stdout.shouldNotContain("line3")
```

Reply:
It's almost impossible to extend the time interval of constant echo-ing so that wait cancellation will hit in the midst of it.

It's irrelevant enough as it is. Having a constant amount of echoes that could not possibly sync with the cancellation is orders of magnitude less relevant.

I'm actually thinking of making SafeStringBuilder internal again 🧢.